File: Cpp/LegalMarket.cpp
#include <iostream>
#include <string>
#include <vector>
#include "Market.cpp"

using namespace std;

class LegalMarket : public Market {
private:
    string LicenseNumber;
    double TaxRate;

public:
    // Default constructor
    LegalMarket() : Market() {
        LicenseNumber = "";
        TaxRate = 0.0;
    }

    // Parameterized constructor
    LegalMarket(int idMarket, string name, string location, string hours,
                string licenseNumber, double taxRate)
        : Market(idMarket, name, location, hours) {
        this->LicenseNumber = licenseNumber;
        this->TaxRate = taxRate;
    }

    // Getter methods
    string getLicenseNumber() {
        return this->LicenseNumber;
    }

    double getTaxRate() {
        return this->TaxRate;
    }

    // Setter methods
    void setLicenseNumber(string licenseNumber) {
        this->LicenseNumber = licenseNumber;
    }

    void setTaxRate(double taxRate) {
        this->TaxRate = taxRate;
    }

    void displayLegalMarket(int indent){
        string sp(indent, ' ');
        cout << sp << "License Number : " << getLicenseNumber() << endl;
        cout << sp << "Tax Rate       : " << getTaxRate() << "%" << endl;
        
        // Display Market attributes
        cout << sp << "Market ID      : " << getIdMarket() << endl;
        cout << sp << "Name           : " << getMarketName() << endl;
        cout << sp << "Location       : " << getMarketLocation() << endl;
        cout << sp << "Operating H.   : " << getOperatingHours() << endl << endl;

        // Display Sellers
        if(!getSellerList().empty()){
            cout << sp << "    Sellers:" << endl;
            for(auto &seller : getSellerList()){
                seller.displaySeller(indent + 8);
            }
        }
    }

    // Destructor
    ~LegalMarket() {
    }
};

File: Java/LegalMarket.java
import java.util.List;

public class LegalMarket extends Market {
    private String licenseNumber;
    private double taxRate;

    // Default constructor
    public LegalMarket() {
        super();
        this.licenseNumber = "";
        this.taxRate = 0.0;
    }

    // Parameterized constructor
    public LegalMarket(int idMarket, String name, String location, String hours,
                       String licenseNumber, double taxRate) {
        super(idMarket, name, location, hours);
        this.licenseNumber = licenseNumber;
        this.taxRate = taxRate;
    }

    // Getter methods
    public String getLicenseNumber() {
        return licenseNumber;
    }

    public double getTaxRate() {
        return taxRate;
    }

    // Setter methods
    public void setLicenseNumber(String licenseNumber) {
        this.licenseNumber = licenseNumber;
    }

    public void setTaxRate(double taxRate) {
        this.taxRate = taxRate;
    }

    // Display method
    public void displayLegalMarket(int indent) {
        String sp = " ".repeat(indent);
        System.out.println(sp + "License Number : " + getLicenseNumber());
        System.out.println(sp + "Tax Rate       : " + getTaxRate() + "%");

        // Display Market attributes
        System.out.println(sp + "Market ID      : " + getIdMarket());
        System.out.println(sp + "Name           : " + getMarketName());
        System.out.println(sp + "Location       : " + getMarketLocation());
        System.out.println(sp + "Operating H.   : " + getOperatingHours());
        System.out.println();

        // Display Sellers
        if (!getSellerList().isEmpty()) {
            System.out.println(sp + "    Sellers:");
            for (Seller seller : getSellerList()) {
                seller.displaySeller(indent + 8);
            }
        }
    }
}
